src/components/genres-list/genres-list.jsx

```diff
@@ -1,21 +1,58 @@
 import React from 'react';
-import {GENRES} from '../../utils/constatns';
-import {genreProp} from '../props/movie-props';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import classNames from 'classnames';
+import {movieProp, genreProp} from '../props/movie-props';
+import {ActionCreator} from '../../store/action';
 
-const Genre = ({genre}) => (
-  <li className="catalog__genres-item">
-    <a href="#" className="catalog__genres-link">{genre}</a>
-  </li>
-);
+const Genre = ({genre, onGenreClick, isCurrentGenre}) => {
+  const handleGenreChange = (evt) => {
+    evt.preventDefault();
+    onGenreClick(genre);
+  };
 
-const GenresList = () => (
-  <ul className="catalog__genres-list">
-    {GENRES.map((genre, i) => <Genre key={i} genre={genre}/>)}
-  </ul>
-);
+  return (
+    <li className={classNames(`catalog__genres-item`, {
+      "catalog__genres-item--active": isCurrentGenre
+    })}>
+      <a href="#" className="catalog__genres-link" onClick={handleGenreChange}>{genre}</a>
+    </li>
+  );
+};
+
+const GenresList = ({currentGenre, movies, onGenreClick}) => {
+  const allGenres = movies.map(({genre}) => genre);
+  const uniqueGenres = [`All genres`, ...new Set(allGenres)];
+
+  return (
+    <ul className="catalog__genres-list">
+      {uniqueGenres.map((genre, i) => <Genre key={i} genre={genre} onGenreClick={onGenreClick} isCurrentGenre={currentGenre === genre}/>)}
+    </ul>
+  );
+};
+
+const mapStateToProps = (state) => ({
+  currentGenre: state.genre,
+  movies: state.movies
+});
```

Comment:
```suggestion
const mapStateToProps = ({genre, movies }) => ({
  genre,
  movies
});
```

currentGenre - зачем тебе так? Переименуй в genre